includes/Modules/Analytics_4.php

```
@@ -689,6 +701,76 @@ protected function create_data_request( Data_Request $data ) {
 				$parent = self::normalize_property_id( $data['propertyID'] );
 
 				return $this->get_analyticsadsenselinks_service()->properties_adSenseLinks->listPropertiesAdSenseLinks( $parent );
+			case 'GET:audiences':
+				$settings = $this->get_settings()->get();
+				if ( empty( $settings['propertyID'] ) ) {
+					return new WP_Error(
+						'missing_required_setting',
+						__( 'No connected Google Analytics 4 property ID.', 'google-site-kit' ),
+						array( 'status' => 500 )
+					);
+				}
+
+				$analyticsadmin = $this->get_analyticsaudiences_service();
+				$property_id    = self::normalize_property_id( $settings['propertyID'] );
+
+				return $analyticsadmin
+					->properties_audiences
+					->listPropertiesAudiences( $property_id );
+			case 'POST:create-audience':
+				$settings = $this->get_settings()->get();
+				if ( ! isset( $settings['propertyID'] ) ) {
+					return new WP_Error(
+						'missing_required_setting',
+						__( 'No connected Google Analytics 4 property ID.', 'google-site-kit' ),
+						array( 'status' => 500 )
+					);
+				}
+
+				if ( ! isset( $data['audience'] ) ) {
+					return new WP_Error(
+						'missing_required_param',
+						/* translators: %s: Missing parameter name */
+						sprintf( __( 'Request parameter is empty: %s.', 'google-site-kit' ), 'audience' ),
+						array( 'status' => 400 )
+					);
+				}
```

We need to throw the `Missing_Required_Param_Exception` exception.